src/services/projectService.ts

```diff
@@ -35,17 +49,15 @@ export default class ProjectService implements IProjectService {
                     project.id = shortid.generate();
                 }
 
-                const storageProvider = StorageProviderFactory.create(
-                    project.targetConnection.providerType,
-                    project.targetConnection.providerOptions,
-                );
-
+                const storageProvider = StorageProviderFactory.createFromConnection(project.targetConnection);
                 await this.saveExportSettings(project);
+                project = encryptProject(project, securityToken);
                 await this.saveProjectFile(project);
 
                 await storageProvider.writeText(
                     `${project.name}${constants.projectFileExtension}`,
-                    JSON.stringify(project, null, 4));
+                    JSON.stringify(project, null, 4),
```

Comment:
I know this wasn't directly from this PR, but what's 4 here? Any way we can name that variable?

Reply:
default indenting when generating the JSON string.  Allows the project file to be written with indents.